x-pack/solutions/search/plugins/enterprise_search/server/deprecations/index.ts

```diff
@@ -0,0 +1,404 @@
+import { DeprecationsDetails } from '@kbn/core-deprecations-common';
+import { GetDeprecationsContext, RegisterDeprecationsConfig } from '@kbn/core-deprecations-server';
+
+import { hasFleetServers } from '@kbn/fleet-plugin/server/services/fleet_server';
+import { isAgentlessEnabled } from '@kbn/fleet-plugin/server/services/utils/agentless';
+import { i18n } from '@kbn/i18n';
+import { Connector, fetchConnectors } from '@kbn/search-connectors';
+
+import { ConfigType } from '..';
+
+export const getRegisteredDeprecations = (
+  config: ConfigType,
+  cloud: CloudSetup,
+  docsUrl: string
+): RegisterDeprecationsConfig => {
+  return {
+    getDeprecations: async (ctx: GetDeprecationsContext) => {
+      const hasAgentless = isAgentlessEnabled();
+      const hasFleetServer = await hasFleetServers(
+        ctx.esClient.asInternalUser,
+        ctx.savedObjectsClient
+      );
+      return [
+        ...getEnterpriseSearchNodeDeprecation(config, cloud, docsUrl),
+        ...(await getCrawlerDeprecations(ctx, docsUrl)),
+        ...(await getNativeConnectorDeprecations(
+          ctx,
+          hasAgentless,
+          hasFleetServer,
+          cloud,
+          docsUrl
+        )),
+      ];
+    },
+  };
+};
+
+/**
+ * If Enterprise Search Node is configured, it's marked as a critical deprecation
+ * Warns that removing the node will disable crawlers/connectors
+ */
+export function getEnterpriseSearchNodeDeprecation(
+  config: ConfigType,
+  cloud: CloudSetup,
+  docsUrl: string
+): DeprecationsDetails[] {
+  if (config.host) {
+    const steps = [];
+    let addendum: string = '';
+    const isCloud = !!cloud?.cloudId;
+    if (isCloud) {
+      steps.push(
+        ...[
+          i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.gotocloud', {
+            values: { baseUrl: cloud.baseUrl },
+            defaultMessage:
+              'Go to {baseUrl} and select this deployment. Or click the link above to go straight there.',
+          }),
+          i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.clickedit', {
+            defaultMessage: "Click the 'Edit' tab",
+          }),
+          i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.scrolldown', {
+            defaultMessage: "Scroll down to the 'Enterprise Search' section",
+          }),
+          i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.clickx', {
+            defaultMessage: "Click the red 'X' by 'Enterprise Search instances'",
+          }),
+          i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.nocapacity', {
+            defaultMessage: 'You should no longer see any Enterprise Search capacity',
+          }),
+          i18n.translate('xpack.enterpriseSearch.deprecations.entsearchhost.clicksave', {
+            defaultMessage: "Click 'Save' and confirm",
+          }),
+        ]
+      );
+      addendum = `\n\n[Click here to manage your deployment](${
```

Comment:
why is the `addendum` not translated if the rest of the steps are?

Reply:
ah, it's not a "step". I wanted to have this part of the description be different, depending on context.

Maybe I just need to give it is own `i18n.translate`call 🤔 I'll do that.